pkg/controller/handlers/retention/retention.go

```diff
@@ -0,0 +1,72 @@
+package retention
+
+import (
+	"time"
+
+	"github.com/obot-platform/nah/pkg/router"
+	"github.com/obot-platform/obot/logger"
+	v1 "github.com/obot-platform/obot/pkg/storage/apis/obot.obot.ai/v1"
+	"k8s.io/apimachinery/pkg/fields"
+	kclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var log = logger.Package()
+
+type Manager struct {
+	c      kclient.Client
+	policy time.Duration
+}
+
+func NewRetentionManager(c kclient.Client, policy time.Duration) *Manager {
+	if policy == 0 {
+		log.Infof("retention policy: disabled")
+	} else {
+		log.Infof("retention policy: %s", policy)
+	}
+
+	return &Manager{
+		c:      c,
```

Comment:
As I understand it, the only reason to pass this client is so that you can make the list request below with multiple field selectors. Since that would be an uncached list, I don't think it's worth doing this because this handler will be called a lot.

Reply:
fixed